samplers/src/main/java/io/opentelemetry/contrib/sampler/internal/CelBasedSamplerComponentProvider.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.sampler.internal;
+
+import dev.cel.common.CelValidationException;
+import io.opentelemetry.api.incubator.config.DeclarativeConfigException;
+import io.opentelemetry.api.incubator.config.DeclarativeConfigProperties;
+import io.opentelemetry.contrib.sampler.CelBasedSampler;
+import io.opentelemetry.contrib.sampler.CelBasedSamplerBuilder;
+import io.opentelemetry.sdk.autoconfigure.spi.internal.ComponentProvider;
+import io.opentelemetry.sdk.extension.incubator.fileconfig.DeclarativeConfiguration;
+import io.opentelemetry.sdk.trace.samplers.Sampler;
+import java.util.List;
+
+/**
+ * Declarative configuration SPI implementation for {@link CelBasedSampler}.
+ *
+ * <p>This class is internal and is hence not for public use. Its APIs are unstable and can change
+ * at any time.
+ */
+public class CelBasedSamplerComponentProvider implements ComponentProvider<Sampler> {
+
+  private static final String ACTION_RECORD_AND_SAMPLE = "RECORD_AND_SAMPLE";
+  private static final String ACTION_DROP = "DROP";
+
+  @Override
+  public Class<Sampler> getType() {
+    return Sampler.class;
+  }
+
+  @Override
+  public String getName() {
+    return "cel_based";
+  }
+
+  @Override
+  public Sampler create(DeclarativeConfigProperties config) {
+    DeclarativeConfigProperties fallbackModel = config.getStructured("fallback_sampler");
+    if (fallbackModel == null) {
+      throw new DeclarativeConfigException(
+          "cel_based sampler .fallback_sampler is required but is null");
+    }
+    Sampler fallbackSampler;
+    try {
+      fallbackSampler = DeclarativeConfiguration.createSampler(fallbackModel);
+    } catch (DeclarativeConfigException e) {
+      throw new DeclarativeConfigException(
+          "cel_based sampler failed to create .fallback_sampler sampler", e);
+    }
```

Comment:
Factoring this out to a method called `getFallbackModel()` could probably help to shrink this long method and improve readability. 👍🏻